### PR TITLE
week 2 MVP pull request

### PR DESCRIPTION
**Pull request by peter0083**
- working on resolve CUDA and CuDNN issue on AWS EC2 Ubuntu instance.

```
RuntimeError: cuDNN error: CUDNN_STATUS_NOT_INITIALIZED
terminate called after throwing an instance of 'c10::Error'
```

+-----------------------------------------------------------------------------+
| NVIDIA-SMI 410.104      Driver Version: 410.104      CUDA Version: 10.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla K80           On   | 00000000:00:1E.0 Off |                    0 |
| N/A   37C    P8    30W / 149W |      0MiB / 11441MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+

NAME="Ubuntu"
VERSION="16.04.5 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04.5 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
VERSION_CODENAME=xenial
UBUNTU_CODENAME=xenial

What I'm planning to do:
reinstall Nvidia driver from the public Nvidia driver page

### CODE REVIEW

**Review by JMHOO — COMMENTED** [review]
Should remove all these `dot` files.

File: src/attngan/code/datasets.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+from PIL import Image
+import numpy.random as random
+if sys.version_info[0] == 2:
+    import cPickle as pickle
+else:
+    import pickle
+
+
+def prepare_data(data):
+    imgs, captions, captions_lens, class_ids, keys = data
```
Comment (JMHOO) [inline src/attngan/code/datasets.py:39]: Add docstring about what's in `data`

File: src/attngan/code/main.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+import dateutil.tz
+import argparse
+import numpy as np
+
+import torch
+import torchvision.transforms as transforms
+
+os.environ['KMP_DUPLICATE_LIB_OK']='True'
+
+dir_path = (os.path.abspath(os.path.join(os.path.realpath(__file__), './.')))
```
Comment (JMHOO) [inline src/attngan/code/main.py:22]: Would recommend use `PYTHONPATH` in this case

**Comment by JMHOO** [comment]
Suggest remove the data files out of the PR, at least use individual PR to include data files. If use the public dataset, write a bash script or point out how to download in the readme is fine.

File: src/attngan/code/main.py (hunk 2 above)
Comment (peter0083) [reply src/attngan/code/main.py:22]: will explore `PYTHONPATH` for future projects. Thanks for the pro tip!

File: src/attngan/code/datasets.py (hunk 1 above)
Comment (peter0083) [reply src/attngan/code/datasets.py:39]: - doctstring added
- `dot` files removed

**Comment by peter0083** [comment]
@JMHOO I have made changes according to your recommendation. If it looks good, please merge. Thanks again!

**Comment by JMHOO** [comment]
I still can see some dot files, such as:
src/attngan/.DS_Store
src/attngan/DAMSMencoders/.DS_Store
src/attngan/code/.DS_Store

besides that, looks good to me.

You can merge into master by yourself. And since you have 126 commits in this PR, I would suggest to use Squash and Merge to reduce the history.

**Comment by peter0083** [comment]
- deleted the remaining `dot` files as @JMHOO pointed out
- used merge and squash to reduce the commit histories. now merging pull request.

**Pull request merged**